erpnext/stock/doctype/delivery_note/delivery_note.py

```diff
@@ -706,3 +723,47 @@ def email_coas(docname):
 	)
 
 	return "success"
+
+def validate_delivery_window(doc, method):
```

Comment:
If this was originally a hook, put this in the DN controller

Reply:
Can't, its called in multiple doctypes.